### DISCUSSION TITLE
Integrazione SONOFF mini Zigbee in MQTT tramite ZigBee2MQTT e ConBee II

### DISCUSSION THREAD

**Discussion by francescobuonfigliodomotica** [indomus/forum#769] [ZigBee, Z-Wave, Thread, Matter]
Ho preso un po' di elementi della famiglia SONOFF Zigbee e li sto integrando ad hass.io tramite MQTT. Tutti quelli a batteria l'interfaccia di Zigbee2MQTT li vede immediatamente. Il Mini non c'è verso. Ne ho montati due nella stessa stanza alla distanza di tre metri dalla Conbee 2 ma proprio non riesco ad integrarli. Domanda sciocca: va bene il firmware originale o bisogna usare Tasmoda (domanda sciocca perchè non so se Tasmoda si usa anche con Zigbee)?
Grazie anticipatamente a chi vorrà rispondermi.
Francesco

**Comment by snakuzzo** [comment]
Tasmota va sui device con chip ESP826x o ESP32...non sui device Zigbee.
Potresti provare a cambiare canale Zigbee, ma poi ti toccherebbe rifare il pairing di tutti i device.
Io posso assicurarti che i Sonoff ZB Mini funzionano perfettamente con Zigbee2mqtt (provato su più istanze personalmente).
Quindi direi di continuare a provare...e se riesci metti il log di zigbee-herdsman a debug. A tal proposito dai una letta [QUI](https://www.zigbee2mqtt.io/information/debug.html)

**Reply by francescobuonfigliodomotica** [reply]
Gentile come sempre... allora:

- Cambiato canale da 11 a 15 sia nell'interfaccia di Phoscon che nel configuration.yaml di Zigbee2mqtt
- riavviato home assistant
- riavviato deconz
- riavviato Zigbee2mqtt
- staccato e riattaccato il mini
Ci ho messo un'oretta buona (prova e riprova) e l'operazione è andata a buon fine. Ho creato l'elenco delle cose fatte in modo che possa essere d'aiuto a chi incontrerà il mio stesso problema. 
Grazie ancora Snakuzzo, Francesco

**Reply by snakuzzo** [reply]
Figurati...ma perché continui ad usare sia deconz che zigbee2mqtt ?

**Reply by francescobuonfigliodomotica** [reply]
Per ignoranza... perché se è vero che non lo uso, all'interno della guida per l'istallazione di Zigbee2mqtt ad un certo punto mi si dice che devo inserire nella configurazione la voce 'adapter: deconz' . Ero convinto che per Zigbee2mqtt fosse necessario tenere dentro anche deconz. Quindi sta ancora lì anche se non lo uso. Devo dire però che sia ieri che oggi se l'integrazione deconz non è avviata, ho problemi con Zigbee2mqtt. Dici che posso eliminare sia l'integrazione che l'add-on?

**Reply by snakuzzo** [reply]
ma noooo :D
Quella configurazione che aggiungi serve solo a dire a zigbee2mqtt che stai usando la conbee (come trovi anche sulla [documentazione ufficiale](https://www.zigbee2mqtt.io/information/supported_adapters.html)) :)

![capture_2021-09-26_18-40](https://user-images.githubusercontent.com/7205371/134816363-57b1c7f2-237f-4e9b-9c86-ed892573446b.png)

**Reply by francescobuonfigliodomotica** [reply]
Ok, allora posso eliminare tutto quello che riguarda deconz! 
Grazie ancora , snakuzzo, alla prossima.
Francesco

**Comment by andreabaroni** [comment]
Buonasera, io mi trovo in una situazione molto simile...anzichè avere un conbee però utilizzo la chiavetta di Itead cc2531.
Dopo un giorno intero, ancora non riesco a trovare una soluzione per far accoppiare il sonoff Mini (ne ho installati 2 per essere sicuro che non fosse uno difettoso).
MQTT Broker configurato
Zigbee2MQTT configurato 
HASSIO funzionante e riavviato più volte dopo ogni installazione dei vari add-on

Consigli su cosa fare ?

EDIT Entrambi i MINI lampeggiano in modo costante (1 led al secondo) e anche tenendo premuto il tasto RESET per 5/10 secondi, non ci sono cambi di stato del led

**Reply by snakuzzo** [reply]
Devi semplicemente riprovare...nella peggiore delle ipotesi puoi provare a cambiare canale, però in questo caso dovrai riaccoppiare tutti gli altri device

**Reply by andreabaroni** [reply]
Ho già provato a cambiare canale ma nulla...Utilizzo SSD come storage su Pi4, ho letto che può provocare delle interferenze, così ho preso un cavo per allontanare l'SSD il più possibile. Continuerò a provare.

**Reply by andreabaroni** [reply]
Mi sono dimenticato di aggiornarvi sul mio problema, sperando possa essere utile in chi si trova in questa situazione.
Sistema:
Chiavetta CC2531 (nel mio caso già flashata da ITEAD, consiglio l'acquisto se non volete passare tramite debugger)
HA installato su Raspberry Pi4, con l'utilizzo di un SSD (che sembra anche essere il fulcro del problema).

Problema:
Provando ad accoppiare ZBMINI non si riesce in alcun modo a riconoscerli.

Soluzione:
ReFLASH :Potrebbe essere un problema di Flash della chiavetta, ma se oltre agli ZBMINI avete altri dispositivi e vengono riconosciuti, non è questa la soluzione.
Cambiare canale: Volendo provare, visto che la rete 2.4Ghz è molto intasata dal wifi, cambiare canale potrebbe risolvere il problema...nel mio caso non ha funzionato, ma tentar non nuoce.
Prolunga USB: Pensavo di essere arrivato alla frutta, invece con 1,99€ ho risolto il problema. Le antenne integrate delle CC2531 sono ahimè molto deboli e per qualche ragione gli SSD interferiscono con esse disturbandole non poco. Utilizzando una prolunga possiamo allontanare il più possibile la nostra chiavetta dal sistema e darle un pò di aria pulita dalla quale captare i vari segnali. Fidatevi FUNZIONA.

Se invece volete sperimentare il saldatore, potete comprare un'antenna ausiliaria e collegarla con una saldatura alla chiavetta.

**Reply by snakuzzo** [reply]
Questa cosa della prolunga si sa da anni ormai

**Reply by andreabaroni** [reply]
Non pensavo di aver scoperto l'acqua calda, ma visto che da neofita della Domotica, mi sono imbattuto in questo problema, e non era stato così facile trovare la soluzione...pensavo potesse esser utile a qualcuno. Se superfluo nessun problema a cancellare il commento.

**Reply by snakuzzo** [reply]
va bene così

**Comment by manny199** [comment]
@francescobuonfigliodomotica Ciao, anche io non riesco a fare il pairing di zbmini con CC2531.
Mi spieghi come hai cambiato canale zigbee in zigbee2mqtt?
Io ho provato aggiungendo in zigbee2mqtt/configuratione.yaml
```
advanced:
  channel: 25
```
Ma non riparte neppure il servizio

Grazie
